stats/stats.go

```diff
@@ -98,7 +102,10 @@ func (s *InPayload) IsClient() bool { return s.Client }
 
 func (s *InPayload) isRPCStats() {}
 
-// InHeader contains stats when a header is received.
+// InHeader contains stats about header reception.
+//
+// - Server-side: The first stats event after the RPC request is received.
+// - Client-side: Occurs after the `OutPayload` event (after the request payload is sent).
```

Comment:
This isn't quite true.  InHeader comes any time after OutHeader.  This is especially true for bidi- or client- streaming RPCs, since OutPayload isn't guaranteed to come right after OutHeader, like it is for unary or server-streaming RPCs.

Maybe exclude the note about client-side here, but you can add a note to `OutHeader` for `Client-side` that says _something like_ `Only occurs after 'Begin', as headers are always the first thing sent on a stream.`